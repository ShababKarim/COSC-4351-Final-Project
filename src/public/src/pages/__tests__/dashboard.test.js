import React, { useState } from 'react';
import { render, cleanup } from '@testing-library/react';
import { renderHook } from '@testing-library/react-hooks';
import Dashboard from '../dashboard';

import AuthContext from '../../context';

afterEach(cleanup);

describe('Dashboard', () => {
	it('renders without crashing', () => {
		const { result } = renderHook(() =>
			useState({
				_id: 1,
				name: 'test',
				email: 'test@gmail.com'
			})
		);
		const change = newObj => (result.current = newObj);
		const { asFragment } = render(
			<AuthContext.Provider
				value={{ auth: result.current, setAuth: change }}
			>
				<Dashboard />
			</AuthContext.Provider>
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
