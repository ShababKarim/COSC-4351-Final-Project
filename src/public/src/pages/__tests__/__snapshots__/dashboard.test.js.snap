// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard renders without crashing 1`] = `
<DocumentFragment>
  <div>
    <div
      class="makeStyles-root-1"
    >
      <header
        class="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionStatic makeStyles-appBar-2 MuiAppBar-colorPrimary"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
        >
          <div
            style="position: relative; width: 40px; height: 40px;"
          >
            <img
              alt="cur"
              src="UH_White_Red.png"
              style="position: absolute; display: block; left: 20px; top: 20px; width: 0px; height: 0px;"
            />
          </div>
          <h5
            class="MuiTypography-root MuiTypography-h5"
            style="float: none; margin-left: auto; margin-right: auto; display: flex;"
          >
            , 
          </h5>
        </div>
      </header>
    </div>
    <div>
      <div
        class="MuiDrawer-root MuiDrawer-docked makeStyles-drawer-75"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation0 MuiDrawer-paper makeStyles-drawerPaper-76 MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft"
        >
          <hr
            class="MuiDivider-root"
          />
          <ul
            class="MuiList-root MuiList-padding"
          >
            <div
              aria-disabled="false"
              class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemIcon-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  role="presentation"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M10.09 15.59L11.5 17l5-5-5-5-1.41 1.41L12.67 11H3v2h9.67l-2.58 2.59zM19 3H5c-1.11 0-2 .9-2 2v4h2V5h14v14H5v-4H3v4c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                  />
                </svg>
              </div>
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary MuiTypography-body1"
                >
                  Logout
                </span>
              </div>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
          </ul>
        </div>
      </div>
    </div>
    <div
      class="makeStyles-page-136"
      id="more"
    >
      <div>
        <ul
          class="MuiGridList-root makeStyles-page-136"
          style="margin: -2px;"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
